### PR TITLE
HTCONDOR-2773: create custom hidden sphinx directive

### PR DESCRIPTION
**Pull request by ColeBollig**
<Insert PR description here, and leave checklist below for code review.>

# HTCondor Pull Request Checklist for internal reviewers

- [x] Verify that (GitHub thinks) the merge is clean. If it isn't, and you're confident you can resolve the conflicts, do so. Otherwise, send it back to the original developer.
- [x] Verify that the related Jira ticket exists and has a target version number and that it is correct.
- [x] Verify that the Jira ticket is in review status and is assigned to the reviewer.
- [x] Verify that the Jira ticket (HTCONDOR-xxx) is mentioned at the beginning of the title. Edit it, if not
- [x] Verify that the branch destination of the PR matches the target version of the ticket
- [ ] Check for correctness of change
- [x] Check for regression test(s) of new features and bugfixes (if the feature doesn't require root)
- [x] Check for documentation, if needed  (documentation [build logs](https://readthedocs.org/projects/htcondor/builds/))
- [x] Check for version history, if needed
- [ ] Check BaTLab dashboard for successful build (https://batlab.chtc.wisc.edu/results/workspace.php) and test for either the PR or a workspace build by the developer that has the Jira ticket as a comment.
- [x] Check that each commit message references the Jira ticket (HTCONDOR-xxx)

## After the above
- Hit the merge button if the pull request is approved and it is not a security patch (security changes require 2 additional reviews)
- If the pull request is approved, take the ticket out of review state
- Assign JIRA Ticket back to the developer

### CODE REVIEW

File: docs/STYLE.md [hunk 1]
```diff
@@ -28,6 +28,117 @@ documentation. Some notable ones are listed below:
 
 ### HTCondor specific markups
 
+**.. hidden::** is a directive to hide sections/portions away in the official ReadTheDocs build while
+still displaying for local builds. This is useful to add documentation for a feature that is not intended
+to be released for a couple versions or to add long winded comments in the docs. Use **.. hidden:: X.Y.Z**
+to specify an intended release version that the documentation should be in the official documentation. If
+the specified version is less than or equal to the current version a warning if printed at build time. You
```
Comment (Todd-L-Miller) [inline docs/STYLE.md:35]:
```suggestion
the specified version is less than or equal to the current version a warning is printed at build time. You
```

File: docs/STYLE.md [hunk 2]
```diff
@@ -28,6 +28,117 @@ documentation. Some notable ones are listed below:
...
+can also use the **:comment:** tag to mark the section as a comment that is never shown (even locally).
+
+Examples:
+
+Hidden section:
+```
+.. hidden::
+
+    This wont build in the ReadTheDocs but will locally!
```
Comment (Todd-L-Miller) [inline docs/STYLE.md:44]:
```suggestion
    This won't build in the ReadTheDocs but will locally!
```

**Review by Todd-L-Miller — APPROVED** [review]
Looks good to me.  Is it intended that you can mark a comment with a version number and get a warning about having not unhidden it?

**Comment by Todd-L-Miller** [comment]
... looks like neither of the test runs even got started? That should be investigated before merging.

**Comment by timtheisen** [comment]
That's because of the ancient CentOS7 host that is the submit node, where the man pages are built for Windows and Mac. Version 2.3.1 of sphinx can't handle it. hidden probably need to be omitted from conf.py.

**Comment by ColeBollig** [comment]
I updated the code to not even do any of the other checks if the `:comment:` tag is used thus squashing any version history warnings.

**Comment by ColeBollig** [comment]
> ... looks like neither of the test runs even got started? That should be investigated before merging.

The older python versions (pre `3.9`) are not happy with the modern type hints (`list[nodes.Node]`) so I reduced the type hint to just `list` and left a comment. It is not ideal, but I would like the hidden capability to be available for the man pages and this should be all it takes to get working (crosses fingers)

**Review by Todd-L-Miller — APPROVED** [review]
The patches look good to me.

**Pull request merged**